cmd/notation/login.go

```diff
@@ -65,6 +67,23 @@ func runLogin(ctx context.Context, opts *loginOpts) error {
 	// initialize
 	serverAddress := opts.server
 
+	// input username and password by prompt
+	reader := bufio.NewReader(os.Stdin)
+	var err error
+	if opts.Username == "" {
+		opts.Username, err = readUsernameFromPrompt(reader)
```

Comment:
If `opts.Username` is read as an empty string with no error, `opts.Password` will be [considered as a token](https://github.com/notaryproject/notation/blob/b7b5fcf56137a0ece3fbc09c608a62860f20a050/cmd/notation/login.go#L123). In that case, should we change the prompt to `Token: ` rather than `Password: `?

Reply:
IMO accepting token is an enhancement and should be done when Notation starts supporting token in cli argument like `-u` and `-p`.

Reply:
I think "Password" is consistent with the --password flag, and also has compliance with container development tools like Docker.

Reply:
> IMO accepting token is an enhancement and should be done when Notation starts supporting token in cli argument like -u and -p.

Notation already supports token if username is empty, although it's reflected in the UX. But I agree that it's an nice-to-have enhancement which is not necessarily to be included in this PR.

Reply:
> I think "Password" is consistent with the --password flag

Yes, not trying to break the UX. Just suggest a more accurate prompt for interactive login. `--password` and `--stdin-password` flags are targeting non-interactive login.

> and also has compliance with container development tools like Docker.

From compliance side, I think Notation only needs to make sure that the generated configuration file is compliant to Docker, not UX. Notation already make changes to login UX, e.g. `docker login` doesn't allow empty username. User must provide empty GUID as username if they want to use a token. Notation allows empty username.